tests/e2e/test_context_test.go

```diff
@@ -922,22 +907,138 @@ func (tc *TestContext) EnsureOperatorInstalledWithGlobalOperatorGroupAndChannel(
+func (tc *TestContext) EnsureOperatorHealthy(nn types.NamespacedName, channel string, targetNamespaces []string) {
+	// Quick health check - if already healthy, we're done
+	if tc.isOperatorHealthy(nn) {
+		return
+	}
+
+	// Not healthy - ensure installation
+	tc.ensureOperatorInstalled(nn, channel, targetNamespaces)
+
+	// Wait for the operator to become healthy
+	tc.ensureCSVSucceeded(nn)
+}
+
+// isOperatorHealthy checks if operator CSV is in Succeeded phase.
+func (tc *TestContext) isOperatorHealthy(nn types.NamespacedName) bool {
+	sub, err := tc.FetchSubscription(nn)
+	if err != nil || sub == nil {
+		return false
+	}
+
+	// Determine CSV name from subscription status (supports both currentCSV and installedCSV)
+	csvName := extractSubscriptionCSVName(sub)
+	if csvName == "" {
+		return false
+	}
+
+	csv, err := fetchResource(
+		tc.NewResourceOptions(
+			WithMinimalObject(gvk.ClusterServiceVersion, types.NamespacedName{
+				Namespace: nn.Namespace,
+				Name:      csvName,
+			}),
+		),
+	)
+
+	if err != nil || csv == nil {
+		return false
+	}
+
+	clusterServiceVersion := &ofapi.ClusterServiceVersion{}
+	tc.convertToResource(csv, clusterServiceVersion)
+
+	return clusterServiceVersion.Status.Phase == ofapi.CSVPhaseSucceeded
+}
 
-	// Ensure the operator's namespace is created.
+// ensureOperatorInstalled performs operator installation, handling pre-existing resources.
+func (tc *TestContext) ensureOperatorInstalled(nn types.NamespacedName, channel string, targetNamespaces []string) {
 	tc.EventuallyResourceCreatedOrUpdated(
 		WithMinimalObject(gvk.Namespace, types.NamespacedName{Name: nn.Namespace}),
-		WithCustomErrorMsg("Failed to create or update namespace '%s'", nn.Namespace),
+		WithCustomErrorMsg("Failed to create namespace '%s'", nn.Namespace),
 	)
 
-	// Ensure the operator group is created or updated.
-	tc.EventuallyResourceCreatedOrUpdated(
-		WithObjectToCreate(tc.createOperatorGroup(nn, []string{nn.Namespace})),
-		WithCustomErrorMsg("Failed to create or update operator group '%s'", resourceID),
+	tc.ensureOperatorGroupExists(nn, targetNamespaces)
+	tc.ensureInstallPlan(nn, channel)
+}
+
+// ensureOperatorGroupExists reuses existing OperatorGroups or creates one.
+func (tc *TestContext) ensureOperatorGroupExists(nn types.NamespacedName, targetNamespaces []string) {
+	existingOG := tc.fetchOperatorGroupInNamespace(nn.Namespace)
+	if existingOG != nil {
+		return // Reuse existing OperatorGroup
+	}
+
+	resourceID := resources.FormatNamespacedName(nn)
+
+	if len(targetNamespaces) == 0 {
+		tc.EventuallyResourceCreatedOrUpdated(
+			WithMinimalObject(gvk.OperatorGroup, nn),
+			WithCustomErrorMsg("Failed to create operator group '%s'", resourceID),
+		)
+	} else {
+		tc.EventuallyResourceCreatedOrUpdated(
+			WithObjectToCreate(tc.createOperatorGroup(nn, targetNamespaces)),
+			WithCustomErrorMsg("Failed to create operator group '%s'", resourceID),
+		)
+	}
```

Comment:
We could remove this if/else:

```suggestion
	tc.EventuallyResourceCreatedOrUpdated(
		WithObjectToCreate(tc.createOperatorGroup(nn, targetNamespaces)),
		WithCustomErrorMsg("Failed to create operator group '%s'", resourceID),
	)
```

Adding an if on the `createOperatorGroup`, to set `targetNamespace` only if passed one have length > 0.